src/encoderstate.c

```diff
@@ -712,7 +712,7 @@ static void encoder_state_encode_leaf(encoder_state_t * const state)
   state->ref_qp = state->frame->QP;
 
   if (cfg->crypto_features) {
-    state->crypto_hdl = kvz_crypto_create();
+    state->crypto_hdl = kvz_crypto_create(&state->encoder_control->cfg);
```

Comment:
When the `kvz_config` structure is passed to `encoder_open` in Kvazaar API, Kvazaar takes a copy of it in the function `kvz_encoder_control_init`. The user can then free or modify the struct without affecting Kvazaar. In order to make sure that the key is available here, it should also be copied in that function instead of using the old pointer. The corresponding free should be added to `kvz_encoder_control_free`.